File: find_player_zero_one_loss.cpp
#include "leetcode.h"

class Solution {
public:
  vvd(int) findWinners(vvd(int) & matches) {
    map<int, int> lost;
    for (auto m : matches) {
      if (!lost.count(m[0]))
        lost[m[0]] = 0;
      lost[m[1]]++;
    }
    vector<int> zero, ones;
    for (auto [k, l] : lost) {
      if (l == 0)
        zero.push_back(k);
      if (l == 1)
        ones.push_back(k);
    }
    return {zero, ones};
  }
};

File: other/find_player_zero_one_loss.rs
use std::collections::BTreeMap;
struct Solution;

impl Solution {
    pub fn find_winners(matches: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        let mut lost: BTreeMap<i32, i32> = matches.into_iter().fold(BTreeMap::new(), |mut hm, m| {
            hm.entry(m[0]).or_insert(0);
            *hm.entry(m[1]).or_insert(0) += 1;
            hm
        });
        return vec![lost.iter().filter(|(_, l)| **l == 0).map(|(&i, _)| i).collect(),
                    lost.iter().filter(|(_, l)| **l == 1).map(|(&i, _)| i).collect()];
    }
}
